extensions/amazon-lambda/runtime/src/main/java/io/quarkus/amazon/lambda/runtime/AmazonLambdaRecorder.java

```diff
@@ -53,12 +55,14 @@ static void initializeHandlerClass(Class<? extends RequestHandler<?, ?>> handler
         ObjectMapper objectMapper = AmazonLambdaMapperRecorder.objectMapper;
         Method handlerMethod = discoverHandlerMethod(handlerClass);
         Class<?> parameterType = handlerMethod.getParameterTypes()[0];
+        Type genericType = handlerMethod.getGenericParameterTypes()[0];
+        JavaType javaType = objectMapper.getTypeFactory().constructType(genericType);
         if (parameterType.equals(S3Event.class)) {
             objectReader = new S3EventInputReader(objectMapper);
         } else {
-            objectReader = new JacksonInputReader(objectMapper.readerFor(parameterType));
+            objectReader = new JacksonInputReader(objectMapper.readerFor(javaType));
         }
-        objectWriter = new JacksonOutputWriter(objectMapper.writerFor(handlerMethod.getReturnType()));
+        objectWriter = new JacksonOutputWriter(objectMapper.writerFor(javaType));
```

Comment:
I'm surprised by this change? It used to use the return type and now AFAICS it uses the parameter type? Is it expected?

Reply:
You are right. To be honest I simply adopted @patriot1burke comment in the issue and didnt gave it more thought. Just wanted to start the process of fixing this.